### PR TITLE
feat(payment): PI-1579 Refactor bluesnapv2 payment strategy in checkout-sdk

### PR DESCRIPTION
**Pull request by bc-ania**
## What?
Refactor the existing `bluesnapv2` payment strategy in Checkout SDK to use the new checkout payment integration API

## Why?
To use the new checkout payment integration API

## Testing / Proof

Screenshot 2024-03-19 at 09 09 45
Screenshot 2024-03-19 at 09 07 56
Screenshot 2024-03-20 at 12 52 18

### CODE REVIEW

File: packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    async initialize(
+        options?: PaymentInitializeOptions & WithBlueSnapV2PaymentInitializeOptions,
+    ): Promise<void> {
+        this._initializeOptions = options && options.bluesnapv2;
+
+        await Promise.resolve(this._paymentIntegrationService.getState());
```
Comment (bc-wpietrzak) [inline packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.ts:82]: Do we need to call `getState` here?

File: packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+        const output = await strategy.execute(getOrderRequestBody());
+
+        await expect(strategy.execute(payload, options)).resolves.toEqual(output);
+    });
+
+    it('throws error is unable to execute due to invalid payment', async () => {
+        await strategy.initialize(initializeOptions);
+        payload.payment = undefined;
+        //     });
```
Comment (vitalii-koshovyi) [inline packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.spec.ts:183]: remove it please

**Review by vitalii-koshovyi — COMMENTED** [review]
Please add testing screenshots

File: packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.ts (hunk 1 above)
Comment (bc-ania) [reply packages/bluesnap-direct-integration/src/bluesnapv2/bluesnapv2-payment-strategy.ts]: I think we don't :) I removed it

**Review by vitalii-koshovyi — APPROVED** [review]

File: packages/bluesnap-direct-integration/src/bluesnapv2/create-bluesnapv2-payment-strategy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { PaymentIntegrationService } from '@bigcommerce/checkout-sdk/payment-integration-api';
+import { PaymentIntegrationServiceMock } from '@bigcommerce/checkout-sdk/payment-integrations-test-utils';
+
+import BlueSnapV2PaymentStrategy from './bluesnapv2-payment-strategy';
+import createBluesnapv2PaymentStrategy from './create-bluesnapv2-payment-strategy';
+
+describe('createBlueSnapDirectApmPaymentStrategy', () => {
```
Comment (bc-wpietrzak) [inline packages/bluesnap-direct-integration/src/bluesnapv2/create-bluesnapv2-payment-strategy.spec.ts:7]: Please correct this name (and the one on line 14) to Bluesnap v2

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**